src/Http/Connection.php

```diff
@@ -214,4 +235,38 @@ protected function shouldRetry(int $statusCode) : bool
     {
         return in_array($statusCode, [401, 403, 408, 429]) || $statusCode >= 500;
     }
+
+    public function tryRefreshHeaderToken(): array
```

Comment:
as @dharitdesai suggested this doesn't belong in this repo. 
May be you should extend this class and create your own retry handler? Or pass the refresh token/access token here so that we can try getting the new access token again